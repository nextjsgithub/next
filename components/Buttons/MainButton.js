import React from 'react';

const MainButton = ({ children, ...props }) => {
  return <button className="MainButton__Btn" {...props}>{children}</button>;
};

export default MainButton;
