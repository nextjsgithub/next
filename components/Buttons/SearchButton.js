import React from 'react';

const SearchButton = ({ children, ...props }) => {
  return <button className="SearchBar__btn" {...props}>{children}</button>;
};

export default SearchButton;
