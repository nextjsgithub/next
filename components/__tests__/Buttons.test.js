import React from 'react';
import { render } from 'enzyme';
import MainButton from '../Buttons/MainButton';
import SearchButton from '../Buttons/SearchButton';

describe('<MainButton /> rendering', () => {
  const wrapper = render(<MainButton />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<SearchButton /> rendering', () => {
  const wrapper = render(<SearchButton />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
