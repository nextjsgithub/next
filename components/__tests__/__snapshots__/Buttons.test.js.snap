// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainButton /> rendering renders correctly 1`] = `
<button
  class="MainButton__Btn"
/>
`;

exports[`<SearchButton /> rendering renders correctly 1`] = `
<button
  class="SearchBar__btn"
/>
`;
